### PR TITLE
Adding in additional logging for realm retrieval to help devs debug JVM cert issues.

### PR DESCRIPTION
**Pull request by jansengsy**
I also added in a change to print out the image full name so devs know where the pull is trying to get the image from should it fail.

### CODE REVIEW

File: galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerRegistryImpl.java [hunk 1]
```diff
@@ -115,8 +118,7 @@ public boolean doYouHave(DockerImageImpl image) {
 		} catch (IllegalStateException e) {
 			return false;
 		} catch (DockerManagerException e) {
-			logger.trace(e);
-			logger.warn("Failed to access registry");
+			logger.warn("Failed to access registry at: " + path);
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerRegistryImpl.java:121]: logger.trace,   and include the stacktrace ",e"

Comment (Michael-Baylis) [reply galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerRegistryImpl.java:121]: failing to access the repository is not an error as such

File: galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerRegistryImpl.java [hunk 2]
```diff
@@ -265,8 +267,18 @@ private boolean retrieveRealm(DockerImageImpl image) throws DockerManagerExcepti
 				}
 			}
 			throw new DockerManagerException("Failed to authenticate, and authentication is required.");
-		} catch (HttpClientException | MalformedURLException | JsonSyntaxException e) {
-			throw new DockerManagerException("Failed to connect to registry", e);
+		} catch (HttpClientException e) {
+			logger.trace("HttpClientException, with message: " + e.getMessage());
+			StringWriter sw = new StringWriter();
+			PrintWriter pw = new PrintWriter(sw);
+			logger.trace("HttpClientException stack trace:\n" + sw.toString());
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerRegistryImpl.java:274]: dont need these traces,     include the information in the throw,  same for all the the others in this try block

File: galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerRegistryImpl.java [hunk 3]
```diff
@@ -265,8 +267,18 @@ private boolean retrieveRealm(DockerImageImpl image) throws DockerManagerExcepti
 				}
 			}
 			throw new DockerManagerException("Failed to authenticate, and authentication is required.");
-		} catch (HttpClientException | MalformedURLException | JsonSyntaxException e) {
-			throw new DockerManagerException("Failed to connect to registry", e);
+		} catch (HttpClientException e) {
+			logger.trace("HttpClientException, with message: " + e.getMessage());
```
Comment (Michael-Baylis) [inline galasa-managers-parent/galasa-managers-cloud-parent/dev.galasa.docker.manager/src/main/java/dev/galasa/docker/internal/DockerRegistryImpl.java:271]: merge this with the throw

**Review by Michael-Baylis — CHANGES_REQUESTED** [review]
some tidy up please

**Comment by jansengsy** [comment]
rerun tests

**Pull request merged**